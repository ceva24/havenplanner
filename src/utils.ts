const createSafeRelativePath = (path: string) => {
    return path.startsWith("/") ? path : "/".concat(path);
};

export { createSafeRelativePath };
