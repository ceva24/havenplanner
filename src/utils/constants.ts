/* eslint-disable import/extensions */
import characterClassData from "@/data/character-classes.json";
import personalQuestData from "@/data/personal-quests.json";

const characterClasses: CharacterClass[] = characterClassData;

const personalQuests: PersonalQuest[] = personalQuestData;

const defaultCharacter: Character = {
    name: "",
    experience: 0,
    gold: 0,
    notes: "",
    characterClass: characterClasses[0],
};

const defaultPersonalQuestCardImage = "/worldhaven/images/personal-quests/gloomhaven/gh-pq-back.png";

export { characterClasses, personalQuests, defaultCharacter, defaultPersonalQuestCardImage };
