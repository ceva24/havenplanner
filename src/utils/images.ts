const baseImageUrl = "https://ceva24.github.io";

const createImageUrl = (relativePath: string) => {
    const safeRelativePath = relativePath.startsWith("/") ? relativePath : `/${relativePath}`;

    return `${baseImageUrl}${safeRelativePath}`;
};

const defaultPersonalQuestCardImageUrl = "/worldhaven/images/personal-quests/gloomhaven/gh-pq-back.jpg";

export { createImageUrl, defaultPersonalQuestCardImageUrl };
