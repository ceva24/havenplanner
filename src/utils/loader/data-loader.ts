import characterClassData from "../../../data/character-classes.json";

const loadCharacterClasses = (): CharacterClass[] => {
    return characterClassData;
};

export { loadCharacterClasses };
