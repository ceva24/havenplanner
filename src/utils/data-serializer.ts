import { characterClasses, initialCharacter } from "@/utils/constants";

interface SerializedCharacterData {
    n: string; // Name
    x: number; // Experience
    g: number; // Gold
    d: string; // Notes
    c: number; // Character id
}

const serialize = (character: Character): string => {
    const characterData: SerializedCharacterData = {
        n: character.name,
        x: character.experience,
        g: character.gold,
        d: character.notes,
        c: character.characterClass.id,
    };

    return JSON.stringify(characterData);
};

const deserialize = (data: string): Character => {
    const characterData = JSON.parse(data) as SerializedCharacterData;

    const character: Character = {
        name: characterData.n,
        experience: characterData.x,
        gold: characterData.g,
        notes: characterData.d,
        characterClass:
            characterClasses.find((characterClass: CharacterClass) => {
                return characterClass.id === characterData.c;
            }) ?? initialCharacter.characterClass,
    };

    return character;
};

export { serialize, deserialize };
