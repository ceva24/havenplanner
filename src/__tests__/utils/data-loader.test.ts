import { loadCharacterClasses } from "../../utils/data-loader";

describe("data loader", () => {
    it("loads character class data", () => {
        const characterClasses = loadCharacterClasses();

        expect(characterClasses).toMatchSnapshot();
    });
});
