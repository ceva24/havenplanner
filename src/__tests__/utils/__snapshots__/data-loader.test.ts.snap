// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data loader loads character class data 1`] = `
Array [
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-brute.png",
    "id": 0,
    "name": "Brute",
  },
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-scoundrel.png",
    "id": 1,
    "name": "Scoundrel",
  },
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-spellweaver.png",
    "id": 2,
    "name": "Spellweaver",
  },
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-tinkerer.png",
    "id": 3,
    "name": "Tinkerer",
  },
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-mindthief.png",
    "id": 4,
    "name": "Mindthief",
  },
  Object {
    "characterMatImageUrl": "/worldhaven/images/character-mats/gloomhaven/gh-cragheart.png",
    "id": 5,
    "name": "Cragheart",
  },
]
`;
